### PR TITLE
chore: update our extension polyfill package

### PR DESCRIPTION
**Pull request by sshanzel**
## Changes
- Since the package has been deprecated since moving from v2 to v3, we have to use the new one.
- Removed comments to check whether some functions are available, should now be supported OOTB.
- Moved the `browser.extension.getURL` to `runtime.getURL`.
- Swapped all imports of the old package to the new one.
- Extension and companion still work.

TODO:
- [x] Fix tests

Previews:
Screenshot 2023-10-26 at 3 40 04 PM
Screenshot 2023-10-26 at 3 39 54 PM
Screenshot 2023-10-26 at 2 01 24 PM

## Events

Did you introduce any new tracking events?
Don't forget to update the [Analytics Taxonomy sheet](https://docs.google.com/spreadsheets/d/18Lv7zXges9QfVX5VYL1a-Hyl0e1sQ3sLr0OK8YZWKXI/edit#gid=0)

Log the new/changed events below:

| Type   | event_name  | value |
|--------|-------------|-------|
| Change/New | event name  | extra: { ... } |

### On those affected packages:
- [ ] Have you done sanity checks in the webapp?
- [ ] Have you done sanity checks in the extension?
- [ ] Does this not break anything in companion?

### Did you test the modified components media queries?
- [ ] MobileL (420px)
- [ ] Tablet (656px)
- [ ] Laptop (1020px)

#### Did you test on actual mobile devices?
- [ ] iOS (Chrome and Safari)
- [ ] Android

WT-1917 #done

### CODE REVIEW

**Review by DragosIorgulescu — COMMENTED** [review]
nice work 🙏  we should be able to merge this directly into main as it's not technically related to manifest v3 -- have you tested the extension?

**Comment by sshanzel** [comment]
Yes, I tested the extension, even though this is just a package upgrade, I am still more inclined to have it in the intermediate branch. Just to make sure we will do the thorough testing phase that will involve the changes here. Wdyt?

@dailydotdev/web-team would love to hear what others think too.

**Comment by DragosIorgulescu** [comment]
> Yes, I tested the extension, even though this is just a package upgrade, I am still more inclined to have it in the intermediate branch. Just to make sure we will do the thorough testing phase that will involve the changes here. Wdyt?
> 
> @dailydotdev/web-team would love to hear what others think too.

sure, I tend to agree, could you rebase this off of `WT-1907-manifest-v3-migration` in that case?

**Review by capJavert — COMMENTED** [review]
As @DragosIorgulescu said, I think rebasing it off w3 epic branch makes sense, looks great 👌

File: packages/extension/webpack.config.js [hunk 1]
```diff
@@ -76,11 +76,6 @@ module.exports = {
 
   resolve: {
     extensions: ['.svg', '.ts', '.tsx', '.js', '.json'],
-    alias: {
-      'webextension-polyfill-ts': path.resolve(
-        path.join(__dirname, 'node_modules', 'webextension-polyfill-ts'),
-      ),
-    },
```
Comment (sshanzel) [inline packages/extension/webpack.config.js:83]: @rebelchris not sure what was the reason behind this alias in the past, but just wanted to tag you to let you know.

**Comment by sshanzel** [comment]
Feel free to approve the PR 😅

**Review by omBratteng — APPROVED** [review]

File: packages/extension/webpack.config.js (hunk 1 above)
Comment (rebelchris) [reply packages/extension/webpack.config.js:83]: Yeah I was just looking at that, but don't know.
Might have been legacy issue for lerna or something.

If everything works let's leave it out.

File: packages/extension/package.json [hunk 2]
```diff
@@ -21,6 +21,7 @@
     "@babel/runtime": "^7.17.9",
     "@dailydotdev/react-contexify": "^5.0.2",
     "@dailydotdev/shared": "*",
+    "@types/webextension-polyfill": "^0.10.5",
```
Comment (rebelchris) [inline packages/extension/package.json:24]: Oh this should be a devDependency no?

File: packages/shared/package.json [hunk 3]
```diff
@@ -111,6 +111,7 @@
     "@growthbook/growthbook": "https://gitpkg.now.sh/dailydotdev/growthbook/packages/sdk-js?a0d1a3add257acdbf5353dfd690547feebbada92",
     "@growthbook/growthbook-react": "^0.17.0",
     "@tippyjs/react": "^4.2.5",
+    "@types/webextension-polyfill": "^0.10.5",
```
Comment (rebelchris) [inline packages/shared/package.json:114]: Same question here

**Review by rebelchris — APPROVED** [review]
Minor question on placement, otherwise looks ok to me.

File: packages/extension/package.json (hunk 2 above)
Comment (sshanzel) [reply packages/extension/package.json:24]: Right. Let me move them.

File: packages/extension/webpack.config.js (hunk 1 above)
Comment (sshanzel) [reply packages/extension/webpack.config.js:83]: Yep. Everything was working including the companion. So we should be safe :shipit:

**Pull request merged**